### PR TITLE
Fix assertion on bvec4 variable initialization

### LINKED ISSUE #836: LLPC crashes on shader that uses bvec4
LLPC errors with:

```
amdllpc: /home/runner/work/gfbuild-llpc/gfbuild-llpc/vulkandriver/drivers/llpc/llpc/translator/lib/SPIRV/SPIRVReader.cpp:3557: llvm::Constant* SPIRV::SPIRVToLLVM::transInitializer(SPIRV::SPIRVValue*, llvm::Type*): Assertion `initializer->getType()->isIntegerTy(1)' failed.
```

on the attached SPIR-V.  I'm attaching an Amber test that has three equivalent shaders - LLPC crashes on the version that declares a bvec4 variable initialized to (false, false, false, false).  Also in the archive is minimized SPIR-V that reproduces the issue.

[tofile.zip](https://github.com/GPUOpen-Drivers/llpc/files/4932668/tofile.zip)

Command used:

```
amdllpc -val=false -gfxip=9.0.0 -verify-ir -auto-layout-desc shader.frag.spv -o shader.frag.spv.out
```

Issue found using spirv-fuzz.

### PR DESCRIPTION
**Pull request by LLJJDD**
This fixes #836

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8281fcb853d636c2bebc4e460a761cad64384046
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 712f76fee75d69b23a1ea8f6465752c3ccaaf9a2</li>
<li>PAL: dev ea5db60841dab7d067f5010f28a980ef222bdf81</li>
<li>SPVGEN: dev 8dc855026f2502ab3f45dadaf0bb802a57d6ad60</li>
<li>XGL: dev 8d9da0426ba869f5f3da23fd72cfaa669ce99c25</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 8abd9486af1727c8c38afcfef7390d70bb7d7d89</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/0)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>No Results</li>

</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>No Results</li>

</ul></details>


</ul></details>

**Comment by LLJJDD** [comment]
> Can you also un-xfail the shaderdb test?

Sure. Done it.

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -3687,10 +3687,11 @@ Constant *SPIRVToLLVM::transInitializer(SPIRVValue *const spvValue, Type *const
     Constant *initializer = cast<Constant>(transValue(spvValue, nullptr, nullptr, false));
     if (initializer->getType() != type) {
       // The translated value type is different to the requested type. This can only happen in the
-      // case that the SPIR-V value was bool but the requested type was i32 because it is a bool
+      // case that the SPIR-V value was (vector of) bool but the requested type was (vector of) i32 because it is a bool
       // in memory.
-      assert(initializer->getType()->isIntegerTy(1));
-      assert(type->isIntegerTy(32));
+      assert(initializer->getType()->isIntegerTy(1) ||
```
Comment (amdrexu) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:3692]: Please use the function isIntOrIntVectorTy(1)

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -3687,10 +3687,11 @@ Constant *SPIRVToLLVM::transInitializer(SPIRVValue *const spvValue, Type *const
...
+             (initializer->getType()->isVectorTy() && initializer->getType()->getContainedType(0)->isIntegerTy(1)));
+      assert(type->isIntegerTy(32) || (type->isVectorTy() && type->getContainedType(0)->isIntegerTy(32)));
```
Comment (amdrexu) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:3694]: The same as above. Use isIntOrIntVectorTy(32)

**Comment by JaxLinAMD** [comment]
retest this please

**Pull request merged**